src/Behat/Behat/Output/Node/Printer/JUnit/JUnitScenarioPrinter.php

```diff
@@ -33,45 +31,27 @@ final class JUnitScenarioPrinter
      */
     private $resultConverter;
 
-    /**
-     * @var JUnitOutlineStoreListener
-     */
-    private $outlineStoreListener;
-
-    /**
-     * @var OutlineNode
-     */
-    private $lastOutline;
-
-    /**
-     * @var int
-     */
-    private $outlineStepCount;
-
     /**
      * @var JUnitDurationListener|null
      */
     private $durationListener;
 
-    public function __construct(ResultToStringConverter $resultConverter, JUnitOutlineStoreListener $outlineListener, JUnitDurationListener $durationListener = null)
+    public function __construct(ResultToStringConverter $resultConverter, JUnitDurationListener $durationListener = null)
```

Comment:
I share some concern about BC. This could also go in hand with [your other comment](https://github.com/Behat/Behat/pull/1459#discussion_r1811575754).
Could it make sense that we release this entire feature in a major version instead?

Reply:
I think we'll need to do a major at some point, but I'm not convinced we should do it now/soon. There's no point releasing a major if the version constraints across the ecosystem will prevent people from upgrading. With the project having been quiet for a while, I think we need to prioritise getting through the backlog on the 3.x series and getting people used to the idea there will be regular releases before we think about what needs a 4.x bump.

That's obviously a different discussion to have elsewhere.

I think there's 4 possibilities:

* Pin this for a major, but it could be waiting a long time
* Release it as-is (apart from fixing the BC break on the constructor) and assume/hope the impact on any reports will be minimal
* Rework it to take a formatter option (as an optional final constructor arg) so it is entirely BC unless people can opt in to the new naming (which could become default in the next major)
* Rework it to a simpler solution for the specific bug, so that we keep the `Some scenario #1` naming but we just ensure the numbering is tied to the feature file sequence, not the test execution. I don't know how feasible that would be though.

My preference is probably to release it with an opt-in formatter option if that can be done without too much work. Or if not then to release it as-is.

What do you / @carlos-granados think?

Reply:
I think that releasing it as is (removing the BC break in the constructor) should be good enough, I think the impact should be minimal

Reply:
Hang on, I just realised this class has been defined final, so no one can extend it and change the constructor signature.

The only problem is if someone does initialize the class with the wrong(old) set of arguments, so here's an idea that preserves BC and shows a deprecation message:
```php
    /**
     * @param null|JUnitDurationListener|JUnitOutlineStoreListener $durationListener
     */
    public function __construct(ResultToStringConverter $resultConverter, $durationListener = null)
    {
        if ($durationListener instanceof JUnitOutlineStoreListener ) {
            @trigger_error('2nd argument is deprecated blabla', E_USER_DEPRECATED);
            $durationListener = func_get_args()[2] ?? null;
        }

        $this->resultConverter = $resultConverter;
        $this->durationListener = $durationListener;
    }
```
wdyt?

Reply:
Yes, it's specifically the concern about external construction (even if it wasn't a final class, the `__construct` method does not enforce signature comparability between children and parents).

Out of the box, behat's error reporting is configured to fail on deprecations so triggering a deprecation is also a BC break IMO.

I would just keep accepting the parameter but leave it unused with a marker to review for the next major. We will probably want to think about how to do deprecations ahead of that anyway (maybe using one of the userland deprecation packages rather than native php errors) and I don't think we should hold this up for that.

Reply:
@acoulton fine by me - I was only wondering what's the best way to do this deprecation or marking, as you named it.

I don't know if we have anything like it, one way could be to make a new issue saying that parameter is now unused and can be removed and tagging it with a milestone.

Just let me know exactly what's the best way forward, I'm fully fine with reverting this part.

Reply:
An issue is a good idea but I think will be too noisy if we have one for each thing. I've added a master issue at #1519 to track small refactorings / removals and added this.

Reply:
So then could just have the constructor as something like:
```
public function __construct(ResultToStringConverter $resultConverter, JUnitOutlineStoreListener $outlineListener, JUnitDurationListener $durationListener = null)
{
    // The unused $outlineListener parameter has been kept temporarily for BC and will be removed in the next major.
    // See https://github.com/Behat/Behat/issues/1519

Reply:
Done: https://github.com/Behat/Behat/pull/1459/commits/576d5eef5a98a9bf0cf87d11ab3c5088e9f92170